File: lab4/lang-tests/test-4/alcance.py
"""
python tiene alcance ESTÁTICO
"""

global_var = 1

def function_a(z):
	print "function_a, z-> ", z
	print "function_a, global_var-> ", global_var
	print "return: ", (global_var+z)
	return (global_var + z)

def function_b(y):
	print "function_b, y-> ", y
	global global_var
	global_var = y+1
	print "function_b, global_var-> ", global_var
	print "function_b, return: ", (global_var * y)
	return function_a(global_var * y)

function_b(3)
"""
terminado
"""

File: lab4/lang-tests/test-4/alcance.scala
object Alcance {
	def main(args: Array[String]) {
		fun_b(3)
	}

	def fun_a( z:Int ) : Int = {
		return z + global_var
	}

	def fun_b( y:Int ) : Int = {
		global_var = y + 1
		return fun_a(y * global_var) 
	}
	
}